### ISSUE TITLE
[DevTools Bug] Cannot remove node "30123" because no matching node was found in the Store.

### ISSUE THREAD

**Issue by joshkaplan** [facebook/react#21755]
### Website or app

https://getstream.github.io/website-react-examples/team/?user=autumn-salad-3&user_token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoiYXV0dW1uLXNhbGFkLTMiLCJleHAiOjE2MjQ2Nzc2MDl9.RgvfWl8k9lsuXIlQ4cM0eCJ6NU8aa1oPiYRe0Pa6zsY&target_origin=https://getstream.io

### Repro steps

1. Go to this [GetStream.io demo app](https://getstream.github.io/website-react-examples/team/?user=autumn-salad-3&user_token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoiYXV0dW1uLXNhbGFkLTMiLCJleHAiOjE2MjQ2Nzc2MDl9.RgvfWl8k9lsuXIlQ4cM0eCJ6NU8aa1oPiYRe0Pa6zsY&target_origin=https://getstream.io)
2. Open React dev tools
3. Send a message in the app
4. Error

### How often does this bug happen?

Every time

### DevTools package (automated)

react-devtools-extensions

### DevTools version (automated)

4.13.5-0ae5290b54

### Error message (automated)

Cannot remove node "30123" because no matching node was found in the Store.

### Error call stack (automated)

```text
at chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:21178:43
    at bridge_Bridge.emit (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:19286:22)
    at chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:19446:12
    at listener (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:37413:39)
```

### GitHub query string (automated)

```text
https://api.github.com/search/issues?q=Cannot remove node  because no matching node was found in the Store. in:title is:issue is:open is:public label:"Component: Developer Tools" repo:facebook/react
```

**Comment by bvaughn** [comment]
Neat app!

I can't reproduce the bug at the URL following the steps you provided, though.

**Comment by Daniel15** [comment]
I hit this issue internally at Facebook too. I can repro in Chrome but not in Firefox.

**Comment by bvaughn** [comment]
@Daniel15 Sharing repro steps with me would be helpful.

**Comment by Daniel15** [comment]
@bvaughn I posted in the Workplace group with a repro :) 
--
Regards,
Daniel Lo Nigro
https://d.sb[https://d.sb/]

Sent from my phone

Jul 14, 2021 8:54:43 AM Brian Vaughn ***@***.***>:

**Comment by sfleiter** [comment]
Any news?
This is happening quite often in main thread overload scenrios for me.

**Comment by bvaughn** [comment]
If you're able to repro this, @sfleiter, you might be in a good place to potential investigate and share ideas for a fix. In case it's not clear from the thread above, I have not been able to reproduce it myself.

**Comment by ShivamS136** [comment]
This and the other bug [21755](https://github.com/facebook/react/issues/21755) seem related as I started to face both together.

**Comment by afreen23** [comment]
I hit this today in firefox.
Something like `Cannot remove node "6809" because no matching node was found in the Store.`

**Comment by varaprasadh2** [comment]
it's happening for me as well,  nothing wrong with my code, but getting this quite often. 
![Screenshot from 2021-09-15 14-49-34](https://user-images.githubusercontent.com/90190222/133406863-fd362b3d-2341-4479-b6fb-67bf5b408d42.png)

**Comment by bvaughn** [comment]
Hey folks 👋🏼  Leaving "me too" comments aren't very helpful. What we need is a way to reproduce this bug on our end so we can (a) see what's causing it and (b) implement and confirm a fix.

**If you can share a way to reliably reproduce this with us please do**.

**Comment by Mayank-MP05** [comment]
I too faced the Same Bug while working with react-dev-tools but when I clicked reinstalled dev tools its temporarily vanished

**Comment by Liquid2Sky** [comment]
After updating Flipper to Version 127 (coming from 125) I got that error while debugging a react-native app too.
However it seems that only build in version of devtools (`4.22.1`) is affected, switching to global installed version (`4.11.0`) everything works fine.

I am on an Apple M1 MBP and that was often the cause of problems related to react / react-native development 🙃 
@eps1lon @joshkaplan Are you on an M1 too?

**Comment by Alimobasheri** [comment]
Hi, I'm using react-dev-tools plugin in Flipper for debugging react native apps. I'm also using Expo environment with a custom dev client to enable app debugging through Flipper.

The devtools plugin works okay on the first run. But as soon as I reload the app it gives me the following error:
`Uncaught Error: Cannot remove node "0" because no matching node was found in the Store.`

```
The error was thrown at file:///Applications/Flipper.app/Contents/Resources/app.asar/bundle.js:5828:337610
    at Dr.emit (file:///Applications/Flipper.app/Contents/Resources/app.asar/bundle.js:5828:281408)
    at file:///Applications/Flipper.app/Contents/Resources/app.asar/bundle.js:5828:283040
    at file:///Applications/Flipper.app/Contents/Resources/app.asar/bundle.js:5828:686878
    at Array.forEach (<anonymous>)
    at A.oe.onmessage (file:///Applications/Flipper.app/Contents/Resources/app.asar/bundle.js:5828:686861)
    at A.t (file:///Applications/Flipper.app/Contents/Resources/app.asar/bundle.js:5820:3330)
    at A.emit (events.js:315:20)
    at oe.exports.L (file:///Applications/Flipper.app/Contents/Resources/app.asar/bundle.js:5788:14736)
    at oe.exports.emit (events.js:315:20)
    at oe.exports.dataMessage (file:///Applications/Flipper.app/Contents/Resources/app.asar/bundle.js:5788:21208)
    at oe.exports.getData (file:///Applications/Flipper.app/Contents/Resources/app.asar/bundle.js:5788:20405)
    at oe.exports.startLoop (file:///Applications/Flipper.app/Contents/Resources/app.asar/bundle.js:5788:17712)
    at oe.exports._write (file:///Applications/Flipper.app/Contents/Resources/app.asar/bundle.js:5788:17002)
    at doWrite (_stream_writable.js:403:12)
    at writeOrBuffer (_stream_writable.js:387:5)
```

**Comment by bvaughn** [comment]
Fixed by #23156.
